mender-client.sh

```diff
@@ -72,42 +103,43 @@ function auth_request_status() {
 
 # $1 - path to data JSON file for auth request
 function wait_for_authorized() {
+  log "Prepare authorization request"
   # Replace newlines with \n
-  pubkey=$(awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' keys/public.key)
-  hexchars="0123456789ABCDEF"
-  end=$( for i in {1..6} ; do echo -n ${hexchars:$(( $RANDOM % 16 )):1} ; done | sed -e 's/\(..\)/:\1/g' )
+  pubkey=$(awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' "${KEYS_DIR}"/public.key)
 
   # Prepare authorization request
   cat <<- EOF > auth.json
 {
-    "id_data": "{ \"mac\": \"00:11:22${end}\"}",
+    "id_data": "{ \"mac\": \"${MENDER_DEVICE_MAC_ADDRESS}\"}",
     "pubkey": "${pubkey}",
     "tenant_token": "${MENDER_TENANT_TOKEN}"
 }
 EOF
 
-  while true; do
-    echo "Send authorization request."
-    echo "Please authorization the device on the server for it to proceed"
+  log "Send authorization request for '$MENDER_DEVICE_MAC_ADDRESS'"
+  while [ -f "$RUNNING_FILE" ]; do
     status_code=$(auth_request_status)
     if [ "$status_code" == "200" ]; then
-        echo "Client has been authorized"
+        log "Client has been authorized"
       break;
     fi
+    echo -n "."
     sleep 5
   done
 }
 
 function get_jwt() {
   x_men_signature=$(generate_signature)
-  curl -k \
+  curl ${CURL_OPTIONS} -s -k \
     -H "Content-Type: application/json" \
     -H "X-MEN-Signature: ${x_men_signature}" \
     --data "@auth.json" \
     ${MENDER_SERVER_URL}/api/devices/v1/authentication/auth_requests
 }
 
 function send_inventory() {
+  [[ -z "$MENDER_ARTIFACT_NAME" || "$MENDER_ARTIFACT_NAME" == "" ]] && { log "ERROR: MENDER_ARTIFACT_NAME is empty"; exit 1; }
```

Comment:
Same as I commented before
```suggestion
  [[ -z "$MENDER_ARTIFACT_NAME" ]] && { log "ERROR: MENDER_ARTIFACT_NAME is empty"; exit 1; }
```